Sources/Sentry/SentryScope.m

```diff
@@ -476,6 +530,11 @@ - (SentryEvent *__nullable)applyToEvent:(SentryEvent *)event
     return event;
 }
 
+- (void)addObserver:(id<SentryScopeObserver>)observer
+{
+    [self.observers addObject:observer];
+}
```

Comment:
on Java, the observers live in the SentryOptions

Reply:
What's the purpose of them being in the SentryOptions? Is it because you can add custom scope observers? If so, we can discuss adding this for Cocoa, but I would rather keep it right now because the observers are not public, and moving it to SentryOptions requires refactoring. @bruno-garcia, what do you think?

Reply:
one reason is because the sync of scopes come from NDK (Android bits), but its executed by the scope class which is a pure Java class, so adding it to SentryOptions made it easy/possible, the other reason is indeed, custom scope observers would be possible

Reply:
all the cases here are specific to the Android/Java SDK, so I'd be fine the way how it is for iOS, the only downside is custom scopes, maybe @bruno-garcia has opinions

Reply:
I get your point; we can easily change it in the future. I rather keep it simple now.